pkg/windows/windows.go

```diff
@@ -256,23 +248,42 @@ type windows struct {
 	log      logr.Logger
 	// defaultShellPowerShell indicates if the default SSH shell is PowerShell
 	defaultShellPowerShell bool
+	// filesToTransfer is the list of files needed for the windows VM
+	filesToTransfer map[*payload.FileInfo]string
 }
 
 // New returns a new Windows instance constructed from the given WindowsVM
-func New(clusterDNS string, instanceInfo *instance.Info, signer ssh.Signer) (Windows, error) {
+func New(clusterDNS string, instanceInfo *instance.Info, signer ssh.Signer, platform *config.PlatformType) (Windows, error) {
 	log := ctrl.Log.WithName(fmt.Sprintf("wc %s", instanceInfo.Address))
 	log.V(1).Info("initializing SSH connection")
 	conn, err := newSshConnectivity(instanceInfo.Username, instanceInfo.Address, signer, log)
 	if err != nil {
 		return nil, fmt.Errorf("unable to setup VM %s sshConnectivity: %w", instanceInfo.Address, err)
 	}
 
+	files := filesToTransfer
+	if filesToTransfer == nil {
+		srcDestPairs, err := getFilesToTransfer(platform)
+		if err != nil {
+			return nil, fmt.Errorf("unable to get list of files to transfer: %w", err)
+		}
+		files := make(map[*payload.FileInfo]string)
+		for src, dest := range srcDestPairs {
+			f, err := payload.NewFileInfo(src)
+			if err != nil {
+				return nil, fmt.Errorf("could not create FileInfo object for file %s: %w", src, err)
+			}
+			files[f] = dest
+		}
```

Comment:
any reason for moving this out of the getFilesToTransfer() func?

Reply:
Good catch @saifshaikh48. We should refrain from adding inline code to a function especially a top level one.

Reply:
If I moved the fileInfo generation back into the getFilesToTransfer function, I wouldn't be able to unit test, because the NewFileInfo tries to get a real location and a hash for a file that doesn't exist in unit tests. 
I could create another wrapper function maybe?

Reply:
>If I moved the fileInfo generation back into the getFilesToTransfer function, I wouldn't be able to unit test,

This can't be the reason to add a bunch of inline code to a top level function.

>I wouldn't be able to unit test, because the NewFileInfo tries to get a real location and a hash for a file that doesn't exist in unit tests.

I think it is possible to "mock" file creation using a temp file but I am not 100% with your context.

>I could create another wrapper function maybe?

The end result should be a single function call in the `New()` method.

Reply:
I think part of the thing that I'm testing is that it's the right windows path though, which makes pointing to a temp file impossible since it would be checking windows directory paths like C://k, or else I would have to find a way to make the path point to the wrong/temp directory which I feel would defeat the purpose of the test, and involve changing the constants.

Reply:
Understood. But that can't be the reason for you to inline code in a top level function. I am reiterating that the end result should be a single function call in the `New()` method. This doesn't mean that you curtailed to introduce just a single helper method. You can break that helper down too allowing you to unit test the sub-methods.